app/client/keybase/keybase.go

```diff
@@ -54,16 +56,22 @@ type Keybase interface {
 
 // NewKeybase creates a new keybase based on the type and customized with the options provided
 func NewKeybase(conf *configs.KeybaseConfig) (Keybase, error) {
-	switch conf.Type {
-	case types.KeybaseType_FILE:
-		// Open the file-based keybase at the specified path
-		if conf == nil || conf.FilePath == "" {
-			return nil, errors.New("keybase path is required for file-based keybase")
+	if conf == nil {
+		return nil, fmt.Errorf(errInvalidKeybase, "a valid keybase configuration is required")
+	}
+	switch config := conf.Config.(type) {
+	case *configs.KeybaseConfig_File:
+		fileConfig := config.File
+		if fileConfig == nil || fileConfig.Path == "" {
+			return nil, fmt.Errorf(errInvalidKeybase, "keybase path is required for file-based keybase")
 		}
-		return NewBadgerKeybase(conf.FilePath)
-	case types.KeybaseType_VAULT:
-		return hashicorp.NewVaultKeybase(conf)
+		return NewBadgerKeybase(fileConfig.Path)
+
+	case *configs.KeybaseConfig_Vault:
+		vaultConfig := config.Vault
```

Comment:
Should we do some sort of check that the vault config contains the correct parameters here as we do with the badger keybase?